qt.lua

```diff
@@ -575,6 +579,58 @@ function premake.extensions.qt.getQRCDependencies(fcfg)
 
 end
 
+--
+-- Checks if a file is a ts file.
+--
+-- @param filename
+--		The file name to check.
+-- @return
+--		true if the file needs to be run through the lrelease tool, false if not.
+--
+function premake.extensions.qt.isTS(filename)
+	return path.hasextension(filename, { ".ts" })
+end
+
+--
+-- Adds the custom build for a ts file.
+--
+-- @param fcfg
+--		The config for a single file.
+-- @param cfg
+--		The config of the project ?
+--
+function premake.extensions.qt.addTSCustomBuildRule(fcfg, cfg)
+	local qt = premake.extensions.qt
+
+	-- get the project's location (to make paths relative to it)
+	local projectloc = fcfg.project.location
+
+	-- create the output file name
+	local qtqmgenerateddir = fcfg.qtqmgenerateddir or cfg.qtqmgenerateddir
```

Comment:
I would create the output directory (if necessary) here during the generation. See comment below on the `buildcommands` line.

Reply:
See below.

Reply:
What happens if I add `qtqmgenerateddir "foo/bar"` in my project and compile? Does `lrelease` create output folders as necessary, or does it fail?